File: backend/src/main/java/org/example/backend/domain/member/dto/MemberResponseDto.java
package org.example.backend.domain.member.dto;

import org.example.backend.domain.member.entity.Member;

public record MemberResponseDto(
    Long memberId,
    String email,
    String nickname,
    String profileImage,
    Long followerCount,
    Long followingCount
) {
    public MemberResponseDto(Member member, Long followerCount, Long followingCount){
        this(
            member.getMemberId(),
            member.getEmail(),
            member.getNickname(),
            member.getProfileImage(),
            followerCount,
            followingCount
        );
    }
}

File: backend/src/main/java/org/example/backend/domain/member/dto/MemberResponseDto.kt
package org.example.backend.domain.member.dto

import org.example.backend.domain.member.entity.Member

data class MemberResponseDto(
    val memberId: Long?,
    val email: String,
    val nickname: String,
    val profileImage: String?,
    val followerCount: Long?,
    val followingCount: Long?
) {
    constructor(member: Member, followerCount: Long?, followingCount: Long?) : this(
        member.memberId,
        member.email,
        member.nickname,
        member.profileImage,
        followerCount,
        followingCount
    )
}
